apollo-runtime/src/main/java/com/apollographql/apollo/internal/RealApolloStore.kt

```diff
@@ -349,15 +355,19 @@ class RealApolloStore(normalizedCache: NormalizedCache, cacheKeyResolver: CacheK
             cacheKeyResolver(),
             cacheHeaders,
             cacheKeyBuilder)
-        val responseReader = RealResponseReader(operation.variables(), rootRecord,
-            fieldValueResolver, customScalarAdapters, responseNormalizer)
+        val responseReader = RealResponseReader(
+            operation.variables(),
+            rootRecord,
+            fieldValueResolver,
+            customScalarAdapters
+        )
         return try {
-          responseNormalizer.willResolveRootQuery(operation)
           val data = operation.adapter().fromResponse(responseReader, null)
+          val records = operation.normalize(data, customScalarAdapters, networkResponseNormalizer() as ResponseNormalizer<Map<String, Any>?>)
           builder<D>(operation)
               .data(data)
               .fromCache(true)
-              .dependentKeys(responseNormalizer.dependentKeys())
+              .dependentKeys(records.dependentKeys()) // Do we need the dependentKeys here?
```

Comment:
Is there a specific reason why we would need the dependent keys when **reading** from the cache?